### PR TITLE
Provide restart server option in error dialog when gapis lost.

### PR DESCRIPTION
**Pull request by stellama0208**
- Bug: b/159788419.

### CODE REVIEW

File: gapic/src/main/com/google/gapid/Main.java [hunk 1]
```diff
@@ -165,24 +168,44 @@ private void uiStartup(Shell shell) {
         shell.getDisplay().asyncExec(() -> showFirstTimeDialog(shell, models, widgets, onStart));
       }
 
+      // TODO: Only for easy gapis crash testing, will delete before check in.
+      window.statusBar.addListener(SWT.MouseDown, $ -> server.disconnect());
+
```
Comment (stellama0208) [inline gapic/src/main/com/google/gapid/Main.java:173]: For easy testing, a click on the status message composite can trigger a quick gapis losing failure :) This part will be deleted before check in.

File: gapic/src/main/com/google/gapid/MainWindow.java [hunk 2]
```diff
@@ -267,6 +270,62 @@ private Point getDefaultInitialLocation(Point size) {
...
+        }
+      }
+    });
+
+    if (OS.isMac) {
+      MacApplication.init(shell.getDisplay(),
+          () -> showAbout(shell, models.analytics, widgets),
+          () -> showSettingsDialog(shell, models, widgets.theme),
+          file -> models.capture.loadCapture(new File(file)));
+    }
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/MainWindow.java:314]: This doesn't quite work correctly. `MacApplication.init` should only be called once. It adds listeners to the menu each time it is called.

**Comment by pmuetschard** [comment]
We talked f2f to explore an alternative where we do not need to allocate new instances of the `Client`, `Models` and `Widgets` classes, simplifying the code a bit.

**Comment by stellama0208** [comment]
Thanks for the f2f comments, they provide a great help to make this PR better! Those comments are addressed with new commits, ready for another look!

File: gapic/src/main/com/google/gapid/server/Client.java [hunk 3]
```diff
@@ -64,12 +64,18 @@
...
-  private final GapidClient client;
+  private GapidClient client;
+
+  public Client() { /* Initialize an empty wrapper. */}
 
   public Client(GapidClient client) {
     this.client = client;
   }
 
+  public void initializeGapidClient(GapidClient client) {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/server/Client.java:75]: Just call it `setGapidClient`

File: gapic/src/main/com/google/gapid/util/Messages.java [hunk 4]
```diff
@@ -75,6 +75,10 @@
   public static final String SETTINGS_TITLE = "Modify Settings";
   public static final String ERROR_MESSAGE =
       "The application encountered an error:\n%s\n\nPlease check the logs for details.";
+  public static final String SERVER_ERROR_MESSAGE =
+      "\nThe server has exited with an error code of: %s \n" +
+      "Most functions in AGI are unavailable without an server. \n" +
+      "You can restart server, exit, or close the dialog to save your unfinished work.";
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/util/Messages.java:81]:
```suggestion
  public static final String SERVER_ERROR_MESSAGE =
      "\nThe server has exited with an error code of: %s \n" +
      "Most functions in AGI are unavailable without a server. \n" +
      "You can restart the server, exit AGI, or close the dialog to continue without a server.";
```

**Pull request merged**